### ISSUE TITLE
Toggle quotes (`yt`)

### ISSUE THREAD

**Issue by kylechui** [kylechui/nvim-surround#98]
On behalf of #79, add a new keymap for toggling quotes, perhaps `ytq` for "you toggle quotes". For example in the buffer:
```lua
local str = "This seems to be my default example"
```
Hitting `ytq..` will toggle the double quotes `"` &rarr; `'` &rarr; `` ` `` &rarr; `"`. The exact order can probably be determined via some config option in the setup table. Perhaps use aliases or a new section titled `cycles` or something. While the aliases table could be used, it makes some sense to keep it separate, so `dsb` could be preserved for deleting parentheses, while opening up `ytb` for toggling the nearest bracket: `)`, `}`, `]`. Open to suggestions from the community on how they would like the interface to look!

**Comment by kylechui** [comment]
@voyeg3r Seeing as you were the one requesting the feature, perhaps you can give a bit more information/a more concrete description of what you would like to see out of this feature?

**Comment by devkvlt** [comment]
I too would love to have this feature.

**Comment by kylechui** [comment]
@devkvlt Thanks for the input; it will probably be the next "big" feature I add after pattern-based modifications in #101

**Comment by pearofducks** [comment]
Just wanted to throw in that my intuitive guess at how this feature worked would have been that `csqq` would have done what the original author is suggesting `tsq` does.

`csqb` changes quotes to brackets, so it feels somewhat "natural" to me that `csqq` would change the quotes to another quote-type.

**Comment by kylechui** [comment]
@pearofducks Thanks for the kind words :) As for `csqq`, I think this *could* work as a feature but I think it would be relatively limited in scope, at least for me. The example I cited earlier was that I would like `tsb` to toggle between `()` and `{}`, but also `dsb` to delete parentheses (I'm a bit weird), which wouldn't be possible with your suggestion. Also I think that `ts` as a keymap prefix just makes more sense, as it follows the `cs`/`ds` scheme.

**Comment by pearofducks** [comment]
Totally makes sense, just throwing that out there in case it made things easier instead of harder! :)

**Comment by kylechui** [comment]
Yeah, the basic idea is that `ysiwq` doesn't exist, because `q` is ambiguous when it comes to which delimiter pair to add to the buffer. For that reason, `cs)q` or `csqq` wouldn't be "defined behavior" as well.

**Comment by voyeg3r** [comment]
> @voyeg3r Do you think it would be a bad idea to "tie" this feature together with aliases? For example, you would be able to have `tsp` for braces, but that would force `dsp` to delete all sorts of braces as well.
> 
> On second consideration it's probably a bad idea to "mix" the two features together; it's probably better to add a new key `toggle` into the setup table, e.g.
> 
> ```lua
> toggle = {
>     q = { "'", '"', "`" },
> }
> ```

I think having a separated table for toggling quotes is better

**Comment by kylechui** [comment]
Pushing this back a bit since I made a handful of breaking changes here and there for `v2.0.0`, and I would like people to really test what exists first before I add any new features.

**Comment by kylechui** [comment]
Are people still interested in this? Sorry for pushing it back so much :sob:

**Comment by gaoqiangks** [comment]
Love this feature. But I think this feature is language specific since quotes may be escaped, and escape character may be different in languages.
